test/multinode_test.go

```diff
@@ -229,6 +229,81 @@ func TestMultiNodeRefreshToken(t *testing.T) {
 	})
 }
 
+// ensure that token accurately reflects out of band updates
+func TestMultiNodeUpdateAttributes(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+		return
+	}
+
+	_, _, finish := setupMultiNodeTest("TestMultiNodeUpdateAttributes")
+	defer finish()
+
+	c1 := createRTCClient("au1", defaultServerPort, &client.Options{
+		TokenCustomizer: func(token *auth.AccessToken, grants *auth.VideoGrant) {
+			token.SetAttributes(map[string]string{
+				"mykey": "au1",
+			})
+		},
+	})
+	c2 := createRTCClient("au2", secondServerPort, &client.Options{
+		TokenCustomizer: func(token *auth.AccessToken, grants *auth.VideoGrant) {
+			token.SetAttributes(map[string]string{
+				"mykey": "au2",
+			})
+			grants.SetCanUpdateOwnMetadata(true)
+		},
+	})
+	waitUntilConnected(t, c1, c2)
+
+	testutils.WithTimeout(t, func() string {
+		rc2 := c1.GetRemoteParticipant(c2.ID())
+		rc1 := c2.GetRemoteParticipant(c1.ID())
+		if rc2 == nil || rc1 == nil {
+			return "participants could not see each other"
+		}
+		if rc1.Attributes != nil && rc1.Attributes["mykey"] != "au1" {
+			return "rc1's initial attributes are incorrect"
+		}
+		if rc2.Attributes != nil && rc2.Attributes["mykey"] != "au2" {
+			return "rc2's initial attributes are incorrect"
+		}
```

Comment:
Should `rcX.Attributes == nil` be an error too?

Reply:
good point, will update